addons/mail/tests/test_ir_mail_server.py

```diff
@@ -398,46 +416,325 @@ def test_personal_mail_server(self):
+            other_mail = self.env["mail.mail"].with_user(self.user_employee).sudo().create({
+                'state': 'outgoing',
+                'email_to': 'target@test.com',
+                'email_from': self.user_employee.email,
+                'recipient_ids': partners[:7].ids,
+                'headers': '{"test": "test header"}',
+            })
+
+        with self.mock_smtplib_connection(), self.mock_datetime_and_now("2025-01-01 20:37:29"):
+            self.env['mail.mail'].process_email_queue()
+        self.assertEqual(self.mail_server_1.owner_limit_count, TEST_LIMIT)
+        self.assertEqual(other_mail.state, 'outgoing')
+        mails = self.env["mail.mail"].search([('mail_message_id', 'in', (mail.mail_message_id.id, other_mail.mail_message_id.id))])
+        states = mails.mapped('state')
+        self.assertEqual(states.count('sent'), 3)
+        self.assertEqual(states.count('outgoing'), 2)
+        outgoing_1 = mails[0]
+        self.assertFalse(outgoing_1.email_to)
+        self.assertEqual(outgoing_1.state, 'outgoing')
+        self.assertEqual(outgoing_1.create_uid, self.user_employee)
+        self.assertEqual(len(outgoing_1.recipient_ids), 16 - 3 * TEST_LIMIT)  # 1 recipient left
+
+        outgoing_2 = mails[1]
+        self.assertEqual(outgoing_2.email_to, 'target@test.com')
+        self.assertEqual(outgoing_2.state, 'outgoing')
+        self.assertEqual(outgoing_2.create_uid, self.user_employee)
+        self.assertEqual(len(outgoing_2.recipient_ids), 7)
+
+        # The next CRON will send all remaining emails of the first mail (1), and 4 mails for the second one
+        # and schedule the 3 last (7 recipients - 3 sent)
+        with self.mock_smtplib_connection(), self.mock_datetime_and_now("2025-01-01 20:39:29"):
+            self.env['mail.mail'].process_email_queue()
+        self.assertEqual(self.mail_server_1.owner_limit_count, TEST_LIMIT)
+        mails = self.env["mail.mail"].search([('mail_message_id', 'in', (mail.mail_message_id.id, other_mail.mail_message_id.id))])
+        states = mails.mapped('state')
+        self.assertEqual(states.count('sent'), 5)
+        self.assertEqual(states.count('outgoing'), 1)
+
+        outgoing = mails[0]
+        self.assertFalse(outgoing.email_to)
+        self.assertEqual(outgoing.state, 'outgoing')
+        self.assertEqual(outgoing.create_uid, self.user_employee)
+        self.assertEqual(len(outgoing.recipient_ids), 3)
+
+        # Send the last email
+        with self.mock_smtplib_connection(), self.mock_datetime_and_now("2025-01-01 20:42:29"):
+            self.env['mail.mail'].process_email_queue()
+        self.assertEqual(self.mail_server_1.owner_limit_count, 3)
+        mails = self.env["mail.mail"].search([('mail_message_id', 'in', (mail.mail_message_id.id, other_mail.mail_message_id.id))])
+        self.assertEqual(set(mails.mapped('state')), {'sent'})
+
+    @contextmanager
+    def assert_scheduled(self, mails, sent_count, processed_count, now):
```

Comment:
Maybe, to be clearer (+ typing of asserts)
```suggestion
    def assertMailSentThenScheduled(self, mails, to_process_count, sent_count, send_datetime):
```